Spring/src/main/java/ru/andryxa/spring/controllers/BooksController.java

```diff
@@ -0,0 +1,49 @@
+package ru.andryxa.spring.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import ru.andryxa.spring.DTO.BookDTO;
+import ru.andryxa.spring.service.BookService;
+
+import java.util.List;
+
+@RestController
+public class BooksController {
+
+    private final BookService bookService;
+
+    @Autowired
+    public BooksController(BookService bookService) {
+        this.bookService = bookService;
+    }
+
+    @GetMapping("/books")
+    public List<BookDTO> showBooks() {
+        return bookService.getListOfBooks();
+    }
+
+    @PostMapping("/createBook")
+    public BookDTO newBook(@RequestParam("name") String name,
```

Comment:
Поменяй сборку ДТО из нескольких объектов получаемых из параметров запроса в получение объекта целиком из тела запроса. Будет примерно так
```suggestion
    public BookDTO newBook(@RequestParam BookDTO book) {
```